src/controllers/sendResetToken.js

```diff
@@ -0,0 +1,104 @@
+import bcrypt from 'bcrypt';
+import { randomBytes } from 'crypto';
+import user from '../db/models';
+import sendgrid from '../middlewares/helper';
+
+const { User } = user;
+
+/**
+  * @description class representing reset user password
+  *
+  * @class SendResettoken
+  */
+class SendResettoken {
+  /**
+    * @description - This method is responsible for creating creating password reset token
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    * @memberof UserHandler
+    */
+  static async resetToken(request, response) {
+    try {
+      const userData = await User.findOne({
+        where: { email: request.body.email }
+      });
+      if (!userData) {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'An email has been sent to your mail',
+        });
+      }
+      const token = randomBytes(40).toString('hex');
+      const tokenExpiresIn = Date.now() + 3600000; 
+      const userUpdatedData = await userData.update({
+        token,
+        passwordExpiresIn: tokenExpiresIn || userData.token,
+      });
+      const hostUrl = 'http://localhost:3000/reset';
+      const emailResponse = await sendgrid(userData.email, 'noreply@authors.com', hostUrl, token);
+      response.status(200).json({
+        status: 'Success',
+        message: 'Password reset link has been sent to your mail',
+        token: userUpdatedData.token,
+      });
+    } catch (error) {
```

Comment:
Emeka, why are you not returning the error message from the callback?

Reply:
I taught that will expose the database architecture somehow but I now understand why I have to return the specific error I am getting from the query.

Reply:
Good one, Emeka.